src/transcodeImage.js

```diff
@@ -0,0 +1,283 @@
+const dicomCodec = require("@cornerstonejs/dicom-codec");
+const Tags = require("./Tags");
+
+const transcodeOp = {
+  none: 0,
+  decodeOnly: 1,
+  encodeOnly: 2,
+  transcode: 3, // decode and then encode
+};
+
+/**
+ * Static mapping for transcoding image.
+ */
+const transcodeMap = {
+  "1.2.840.10008.1.2": {
+    transferSyntaxUid: "1.2.840.10008.1.2",
+    transcodeOp: transcodeOp.decodeOnly,
+    alias: "uncompressed",
+  },
+  "1.2.840.10008.1.2.1": {
+    transferSyntaxUid: "1.2.840.10008.1.2.1",
+    transcodeOp: transcodeOp.decodeOnly,
+    alias: "uncompressed",
+  },
+  "1.2.840.10008.1.2.2": {
+    transferSyntaxUid: "1.2.840.10008.1.2.1",
+    transcodeOp: transcodeOp.decodeOnly,
+    alias: "uncompressed",
+  },
+  "1.2.840.10008.1.2.4.57": {
+    transferSyntaxUid: "1.2.840.10008.1.2.4.80",
+    transcodeOp: transcodeOp.transcode,
+    alias: "jpeglossless",
+  },
+  "1.2.840.10008.1.2.4.70": {
+    transferSyntaxUid: "1.2.840.10008.1.2.4.80",
+    transcodeOp: transcodeOp.transcode,
+    alias: "jpeglossless",
+  },
+  "1.2.840.10008.1.2.4.90": {
+    transferSyntaxUid: "1.2.840.10008.1.2.4.80",
+    transcodeOp: transcodeOp.transcode,
+    alias: "jp2",
+  },
+  "1.2.840.10008.1.2.4.91": {
+    transferSyntaxUid: "1.2.840.10008.1.2.4.80",
+    transcodeOp: transcodeOp.transcode,
+    alias: "jp2",
+  },
+  "1.2.840.10008.1.2.5": {
+    transferSyntaxUid: "1.2.840.10008.1.2.4.80",
+    transcodeOp: transcodeOp.transcode,
+    alias: "rle",
+  },
+};
+
+/**
+ * Minimum image info data to be used on transcode process by dicom-codec api.
+ */
+function getImageInfo(dataSet) {
```

Comment:
dicom-codec will adapt this set to each codec and this is the minimum set for it.

Reply:
There are more required data elements - you need to know the photometric interpretation.  Note that PMI is the value of the internal encoder, and is NOT the value of the output of the decoder.  The transcoder can change this value, for example transcoding from JPEG-2000 to JPEG-Lossless  will almost always affect the PMI, and both the old and new value are required to be known.  Transcoding to JPEG-Lossless or JPEG-LS with colour images can encode in either RGB or YBR_*, and the YBR versions will be different sizes.  Suggest sticking to RGB.

Reply:
You have made a very good point indeed.

And considering we are talking about transcoding from tJPEG-2000 to JPEG-Lossless. Please, check out the links
openjpeg decoder factory layer
https://github.com/cornerstonejs/codecs/blob/main/packages/openjpeg/src/J2KDecoder.hpp#L293

Charls encoder factory layer
https://github.com/cornerstonejs/codecs/blob/main/packages/charls/src/JpegLSEncoder.hpp#L84
and also here
https://github.com/team-charls/charls/blob/a3066f33d09630a925043c4c30424fe277a251a6/src/charls_jpegls_encoder.cpp#L141

I dont see how anything color is being used/consumed at all.

But now if we check the factory layer for encoding on OpenJpeg
https://github.com/cornerstonejs/codecs/blob/main/packages/openjpeg/src/J2KEncoder.hpp#L228

I can see there is a harded coded color space being passed to encoder.

At end as you can realize this getImageInfo is a tentative of abstracting as much as much the many differences between codecs params. I'm happy to pass down : photometricInterpretation but I dont now how to adapt it to codecs (pmi? I did not find anything)